pallets/did/src/did_details.rs

```diff
@@ -349,10 +352,18 @@ impl<T: Config> DidDetails<T> {
 			Ordering::Less => {
 				let deposit_to_release = self.deposit.amount.saturating_sub(new_required_deposit);
 
-				DidDepositCollector::<T>::free_deposit(Deposit {
-					owner: self.deposit.owner.clone(),
-					amount: deposit_to_release,
-				})?;
+				let is_key_migrated =
```

Comment:
Ah good point. I was thinking since the new required deposit is higher than the current, we can simply hold more balance. But since the amount is getting updated, this is a source of failure. Updated in: [1beb6fe](https://github.com/KILTprotocol/kilt-node/pull/540/commits/1beb6fef6b99cab2abcfcfe50283b1c970dd584f)